### PR TITLE
redux setup, login, forgot password, about us page

### PR DESCRIPTION
**Pull request by p-anja**

### CODE REVIEW

File: moja-kuhinjica-frontend/src/reduxStore/reducers/userReducer.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { Reducer } from '@reduxjs/toolkit'
+import { ActionTypes } from '../constants/actionTypes'
+
+interface UserState {
+    user: {
+        id: number | null
+        name: string | null
+        surname: string | null
+        phoneNumber: string | null
+        role: string | null
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/reduxStore/reducers/userReducer.ts:10]: Add in progress and error handling

**Review by koskeee — CHANGES_REQUESTED** [review]
Checkup comments

File: moja-kuhinjica-frontend/src/pages/login/index.tsx [hunk 2]
```diff
@@ -23,17 +24,10 @@ const LoginPage = (): JSX.Element => {
...
-            .then((res) => {
-                localStorage.setItem('token', res.data.access_token)
-                router.push('/')
-                reset()
-            })
-            .catch((err) => {
-                setShowError(true)
-                console.log(err)
-            })
+        dispatch(userLogin(inputData))
```
[the remark this answers is not in the document]
Comment (p-anja) [reply moja-kuhinjica-frontend/src/pages/login/index.tsx]: Done.

File: moja-kuhinjica-frontend/src/reduxStore/reducers/userReducer.ts (hunk 1 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/reduxStore/reducers/userReducer.ts:10]: Done.

File: moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
 import { ErrorLabel } from '@/components/label/ErrorLabel'
+import { Text } from '@/components/label/Text'
 import { bgModal } from '../../../constants/constants'
 import styles from './LoginModal.module.scss'
-import email from '../../../../public/static/assets/images/email.svg'
-import password from '../../../../public/static/assets/images/password.svg'
-import { Text } from '@/components/label/Text'
+import email from 'public/static/assets/images/email.svg'
+import password from 'public/static/assets/images/password.svg'
+import loadingIcon from 'public/static/assets/images/loading.png'
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx:15]: You should add spinning icon to the Regular Button component and add it here

File: moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                        Zaboravili ste šifru?
+                    </label>
+                    <Text
+                        content="Ne brinite, mi ćemo Vam poslati instrukcije za resetovanje."
+                        style={styles.infoLabel}
+                    />
+                    {showError && <ErrorLabel content={errorMessage} />}
+                    <FormInput
+                        register={register}
+                        errors={errors}
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx:74]: Add validation for email

File: moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react'
+import Modal from 'react-modal'
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx:2]: Add forgot and reseting modal mobile version

File: moja-kuhinjica-frontend/src/pages/index.tsx [hunk 6]
```diff
@@ -11,27 +11,41 @@ import { SignUpModal } from '@/components/modal/signUp/SignUpModal'
...
+
+    const dispatch = useAppDispatch()
+    const isAuthorized = useAppSelector((state) => state.auth.isAuthorized)
     const router = useRouter()
     const ref = useRef<HTMLDivElement>(null)
 
     useEffect(() => {
-        isLoggedIn()
-    }, [])
+        if (isAuthorized) dispatch<any>(setLoggedInUser())
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/pages/index.tsx:47]: Maybe name should be loadUser

File: moja-kuhinjica-frontend/src/reduxStore/actions/userActions.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        type: ActionTypes.SET_LOGGED_IN_USER,
+        payload: data,
+    })
+}
+
+export const userLogin =
+    ({ inputData, onSuccess, onError }: any) =>
+    async (dispatch: any) => {
+        dispatch({
+            type: ActionTypes.USER_LOGIN_IN_PROGRESS,
```
Comment (koskeee) [inline moja-kuhinjica-frontend/src/reduxStore/actions/userActions.ts:16]: Rename it  to USER_LOGIN

**Review by koskeee — CHANGES_REQUESTED** [review]
Check comments

File: moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx (hunk 4 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx:74]: Done.

File: moja-kuhinjica-frontend/src/pages/index.tsx (hunk 6 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/pages/index.tsx:47]: Done.

File: moja-kuhinjica-frontend/src/reduxStore/actions/userActions.ts (hunk 7 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/reduxStore/actions/userActions.ts:16]: Done.

File: moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx (hunk 3 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx:15]: Done.

File: moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx (hunk 5 above)
Comment (p-anja) [reply moja-kuhinjica-frontend/src/components/modal/passwordForgetting/PasswordForgettingModal.tsx:2]: Done.

**Review by koskeee — APPROVED** [review]

**Pull request merged**